spec/jobs/tl3_promotions_spec.rb

```diff
@@ -43,6 +47,26 @@ def create_qualifying_stats(user)
     run_job
   end
 
+  it "allows plugins to control tl3_promotion's promotions" do
+    DiscoursePluginRegistry.register_modifier(plugin, :tl3_custom_promotions, &allow_block)
+    TrustLevel3Requirements.any_instance.stubs(:requirements_met?).never
+    tl2_user = Fabricate(:user, trust_level: TrustLevel[2])
+    create_qualifying_stats(tl2_user)
+    run_job
+  ensure
+    DiscoursePluginRegistry.unregister_modifier(plugin, :tl3_custom_promotions, &allow_block)
+  end
+
+  it "allows plugins to control tl3_promotion's demotions" do
+    DiscoursePluginRegistry.register_modifier(plugin, :tl3_custom_demotions, &array_block)
```

Comment:
One final thing, can you adjust the `array_block` name to something more clear, like `tl3_demotion_block`, and the  `allow_block` to something like `tl3_promotion_block`?